src/onnxruntime.cc

```diff
@@ -339,17 +358,42 @@ ModelState::LoadModel(
       }
 
       // Default GPU execution provider.
-      // Using default values for evrything other than device id and cuda stream
-      OrtCUDAProviderOptions cuda_options{
-          instance_group_device_id,
-          OrtCudnnConvAlgoSearch::EXHAUSTIVE,  // cudnn_conv_algo_search
-          std::numeric_limits<size_t>::max(),  // gpu_mem_limit
-          0,                                   // arena_extend_strategy
-          true,                                // do_copy_in_default_stream
-          stream != nullptr ? 1 : 0,
+      // Using default values for everything other than device id and cuda
+      // stream
+      OrtCUDAProviderOptions cuda_options;
+      cuda_options.device_id = instance_group_device_id;
+      cuda_options.cudnn_conv_algo_search = OrtCudnnConvAlgoSearch::EXHAUSTIVE;
+      cuda_options.gpu_mem_limit = std::numeric_limits<size_t>::max();
+      cuda_options.arena_extend_strategy = 0;
+      cuda_options.do_copy_in_default_stream = true;
+      cuda_options.has_user_compute_stream = stream != nullptr ? 1 : 0;
+      cuda_options.user_compute_stream =
           stream != nullptr ? (void*)stream : nullptr,
-          nullptr  // default_memory_arena_cfg
-      };
+      cuda_options.default_memory_arena_cfg = nullptr;
+
+      {
+        // Parse CUDA EP configurations
+        triton::common::TritonJson::Value params;
+        if (model_config_.Find("parameters", &params)) {
+          int cudnn_conv_algo_search = 0;
+          RETURN_IF_ERROR(TryParseParameter(
+              params, "cudnn_conv_algo_search", &cudnn_conv_algo_search));
```

Comment:
You are assuming 'cudnn_conv_algo_search' is unmodified by this call (and similar below). That is not documented and seems like an unsafe API design. GetParameterValue and underlying calls do not make any guarantees about what happens to that argument when there is an error.

Reply:
I updated the helper functions to use the default value when key is not found. In case of any other error (other than key not found) the code returns. I intentionally do not do any error checking for these values example are the values within allowed range etc because ORT does that... In case the value is not set right then ORT session creation will fail.